dialogue-core/src/main/java/com/palantir/dialogue/core/ContentDecodingChannel.java

```diff
@@ -53,21 +54,43 @@ final class ContentDecodingChannel implements EndpointChannel {
     private static final String CONTENT_ENCODING = "content-encoding";
     private static final String CONTENT_LENGTH = "content-length";
     private static final String GZIP = "gzip";
+    private static final String PREFER_COMPRESSED_RESPONSE_TAG = "prefer-compressed-response";
```

Comment:
Note that I have opted _not_ to add this to the readme quite yet, so we can understand when and why folks opt into it.

Reply:
Thoughts on making this `@VisibleForTesting` so we can use it in the tests when defining `PreferCompressedResponseEndpoint`?

Reply:
I'd actually written that initially, but decided I would prefer not to share the reference because it adds friction if the value changes (the test will fail) and make folks consider the implications for consumers before moving forward. In practice I don't thin it makes a great deal of difference either way, but that was my line of reasoning :-)

Reply:
> it adds friction if the value changes (the test will fail) and make folks consider the implications for consumers before moving forward

I like this reasoning! Sounds good.